### PR TITLE
Improve shape function of `tf.image.draw_bounding_boxes`

### PR DESCRIPTION
**Pull request by yongtang**
The `tf.image.draw_bounding_boxes` requires `boxes` to be 3-D shape though there was no check on shape function.

This fix improves the shape function by restricting the boxes to 3-D.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/core/ops/image_ops.cc [hunk 1]
```diff
@@ -435,6 +435,11 @@ REGISTER_OP("DrawBoundingBoxes")
     .Output("output: T")
     .Attr("T: {float, half} = DT_FLOAT")
     .SetShapeFn([](InferenceContext* c) {
+      ShapeHandle boxes;
+      // boxes: 3-D with shape [batch, num_bounding_boxes, 4]
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 3, &boxes));
+      DimensionHandle unused;
+      TF_RETURN_IF_ERROR(c->WithValue(c->Dim(boxes, 2), 4, &unused));
```
Comment (rmlarsen) [inline tensorflow/core/ops/image_ops.cc]: I'm not sure this is correct. As I read the code, the rank of images should be 4, and the rank of boxes should be 3.
We can improve it further by checking that the depth dimension is either 1, 3, or 4 or unknown. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/draw_bounding_box_op.cc#L48

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by yongtang** [comment]
Thanks @rmlarsen for the review. The PR has been updated. Please take a look.

File: tensorflow/core/ops/image_ops.cc [hunk 2]
```diff
@@ -435,6 +435,25 @@ REGISTER_OP("DrawBoundingBoxes")
...
+      // Channel depth should be either 1 (GRY), 3 (RGB), or 4 (RGBA).
+      if (c->ValueKnown(c->Dim(images, 3))) {
+        int64 depth = c->Value(c->Dim(images, 3));
+        if (!(depth == 1 || depth == 3 || depth == 4)) {
+          return errors::InvalidArgument("Channel depth should be either 1 (GRY), "
+                                         "3 (RGB), or 4 (RGBA)");
+        }
+      }
+
+      // The rank of the boxes should be 3-D: [batch, num_bounding_boxes, 4].
```
Comment (rmlarsen) [inline tensorflow/core/ops/image_ops.cc]: Nit" Either say "the rank of boxes is 3" or "boxes is 3-D".

File: tensorflow/core/ops/image_ops.cc [hunk 3]
```diff
@@ -435,6 +435,25 @@ REGISTER_OP("DrawBoundingBoxes")
     .Output("output: T")
     .Attr("T: {float, half} = DT_FLOAT")
     .SetShapeFn([](InferenceContext* c) {
+      // The rank of the images should be 4.
```
Comment (rmlarsen) [inline tensorflow/core/ops/image_ops.cc]: Please add a unit test in image_ops_test.cc

**Review by rmlarsen — CHANGES_REQUESTED** [review]

File: tensorflow/core/ops/image_ops.cc (hunk 2 above)
Comment (yongtang) [reply tensorflow/core/ops/image_ops.cc]: Thanks @rmlarsen. Done.

File: tensorflow/core/ops/image_ops.cc (hunk 3 above)
Comment (yongtang) [reply tensorflow/core/ops/image_ops.cc]: Thanks @rmlarsen. The test cases have been added.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
Thanks!

**Pull request merged**